### PR TITLE
Remove legacy class OC_Group and OC_User

### PR DESCRIPTION
**Pull request by MorrisJobke**
* basically a straight replacement of the wrapped code at the calling code parts

I did some basic testing, but nearly all of this code is not used anymore and is covered by tests anyway. (or are tests itself)

### CODE REVIEW

**Comment by rullzer** [comment]
did you test with LDAP and all?

File: lib/private/User/User.php [hunk 1]
```diff
@@ -204,12 +204,20 @@ public function delete() {
...
 			// FIXME: Feels like an hack - suggestions?
 
+			$groupManager = \OC::$server->getGroupManager();
+			$userManager = \OC::$server->getUserManager();
 			// We have to delete the user from all groups
-			foreach (\OC::$server->getGroupManager()->getUserGroupIds($this) as $groupId) {
-				\OC_Group::removeFromGroup($this->uid, $groupId);
+			foreach ($groupManager->getUserGroupIds($this) as $groupId) {
+				$group = $groupManager->get($groupId);
+				$user = $userManager->get($this->uid);
```
Comment (nickvergessen) [inline lib/private/User/User.php]: Use `$this` instead of setting the same object up again?

**Review by nickvergessen — DISMISSED** [review]

**Comment by schiessle** [comment]
I would prefer to do such changes at the beginning of a development cycle. This gives us more time for testing and increase the change that we spot problems during development.

File: lib/private/User/User.php (hunk 1 above)
Comment (MorrisJobke) [inline lib/private/User/User.php]: 🙈🙊🙉it was already late

**Comment by MorrisJobke** [comment]
> I would prefer to do such changes at the beginning of a development cycle. This gives us more time for testing and increase the change that we spot problems during development.

Look at the changes and notice that most of it is in not used code paths anymore (old sharing code) or Test code anyway

**Review by rullzer — APPROVED** [review]
I'm fine with this. Still time enough before 12 IMO

File: lib/private/User/User.php (hunk 1 above)
Comment (MorrisJobke) [inline lib/private/User/User.php]: Fixed.

**Comment by MorrisJobke** [comment]
Rebased to fix the merge conflict.

**Review by LukasReschke — APPROVED** [review]

**Comment by MorrisJobke** [comment]
Need to check the failing unit tests

File: lib/private/User/User.php [hunk 2]
```diff
@@ -204,12 +204,18 @@ public function delete() {
...
+				$group = $groupManager->get($groupId);
+				if ($group) {
+					\OC_Hook::emit("OC_Group", "pre_removeFromGroup", ["run" => true, "uid" => $this->uid, "gid" => $groupId]);
+					$group->removeUser($this);
+					\OC_Hook::emit("OC_User", "post_removeFromGroup", ["uid" => $this->uid, "gid" => $groupId]);
+				}
 			}
 			// Delete the user's keys in preferences
-			\OC::$server->getConfig()->deleteAllUserValues($this->uid);
+			$this->config->deleteAllUserValues($this->uid);
```
Comment (MorrisJobke) [inline lib/private/User/User.php]: This breaks the unit tests, because the unit tests juggle around with the global state 🙈

Comment (MorrisJobke) [inline lib/private/User/User.php]: I will revert it, because it would mean to rewrite half of the unit tests :(

**Review by LukasReschke — APPROVED** [review]
:rocket:

**Pull request merged**